pkg/logging/controller/logging/clusterlogging.go

```diff
@@ -0,0 +1,92 @@
+package logging
+
+import (
+	"github.com/sirupsen/logrus"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	loggingconfig "github.com/rancher/rancher/pkg/logging/config"
+	"github.com/rancher/rancher/pkg/logging/generator"
+	"github.com/rancher/rancher/pkg/logging/utils"
+)
+
+type ClusterLoggingSyncer struct {
+	management           config.ManagementContext
+	clusterLoggingClient v3.ClusterLoggingInterface
```

Comment:
We should use *Lister object and fetch objects from cache on Get/List operations. Client should be used only for create/update. Example of lister and its use can be found here: https://github.com/rancher/rancher/blob/master/pkg/cluster/controller/nodesyncer/nodessyncer.go#L37